app/assets/javascripts/angular/controllers/mobile/messages.js

```diff
@@ -0,0 +1,225 @@
+(function() {
+  // Set CSRF Headers to be used for Submitting Messages via Ajax.
+  angular.module('wtsApp').config([
+    "$httpProvider", function($httpProvider) {
+      $httpProvider.defaults.headers.common['X-CSRF-Token'] = $('meta[name=csrf-token]').attr('content');
+    }
+  ]);
+
+  angular.module('wtsApp').controller('messagesCtrl', ['$scope', '$http', '$interval', function($scope, $http, $interval) {
+    // Default state for Messages.
+    $scope.initializeMessages = function() {
+      $scope.resetMessages();
+
+      // Message Filtering
+      $scope.filterOptions = [
+        {"id": 0, "title": "Show all messsages"},
+        {"id": 1, "title": "Show those waiting on me"},
+        {"id": 2, "title": "Show those I am waiting on"}
```

Comment:
Let's change this to: "Show all conversations", "Messages already replied to", "Messages waiting for my reply" or something laong those lines.